request/src/main/java/com/vimeo/networking2/internal/VimeoApiClientImpl.kt

```diff
@@ -140,6 +143,46 @@ internal class VimeoApiClientImpl(
         return vimeoService.createAlbum(authHeader, safeUri, body).enqueue(callback)
     }
 
+    override fun putTeamPermission(
+        permissionPolicyUri: String,
+        teamEntityType: TeamEntityType,
+        teamEntityUri: String,
+        uri: String,
+        callback: VimeoCallback<Unit>,
+        queryParams: Map<String, String>?,
+        bodyParams: Map<String, Any>?
+    ): VimeoRequest {
+
+        val body = bodyParams.intoMutableMap()
+
+        body[ApiConstants.Parameters.PARAMETER_PERMISSION_POLICY_URI] = permissionPolicyUri
+        body[ApiConstants.Parameters.PARAMETER_TEAM_ENTITY_TYPE] = teamEntityType.name
+        body[ApiConstants.Parameters.PARAMETER_TEAM_ENTITY_URI] = teamEntityUri
+
+        val safeUri = uri.validate() ?: return localVimeoCallAdapter.enqueueInvalidUri(callback)
+
+        return vimeoService.put(authHeader, safeUri, queryParams.orEmpty(), body).enqueue(callback)
```

Comment:
We generally prefer making a specific function in the service instead of using the generic `put` function, as it allows us to better define what's going into the body when possible.

Reply:
So why do these exist, and why are they used in other places within the networking library?

Reply:
This `put` function exists exclusively for the `VimeoApiClient.putContent` function, which exists to cover any holes that the library has in API coverage. It is not used anywhere else from what I can see. The same logic applies to the `post` function in `VimeoService`, which is used exclusively by `VimeoApiClient.postContent`. The generic `delete` function is used everywhere because the behavior of a "delete" is normally constant between endpoints since we aren't passing body/query parameters when calling delete.